packages/desktop-client/src/components/accounts/TransactionsTable.js

```diff
@@ -1847,12 +1898,96 @@ export let TransactionTable = forwardRef((props, ref) => {
     id => splitsExpanded.dispatch({ type: 'toggle-split', id }),
     [splitsExpanded.dispatch],
   );
+  let [field, setField] = useState('date');
+  let [ascDesc, setAscDesc] = useState('asc');
+
+  function onSortTable(headerClicked) {
+    if (headerClicked === field) {
+      ascDesc === 'desc' ? setAscDesc('asc') : setAscDesc('desc');
+    } else {
+      setField(headerClicked);
+      headerClicked === 'amount' || headerClicked === 'date'
+        ? setAscDesc('asc')
+        : setAscDesc('desc');
+    }
+  }
+
+  let category = props.categoryGroups.flatMap(cat => {
+    return cat.categories;
+  });
+  let fields = field + 's';
+
+  function sortStrings() {
+    let sortDesc = transactions
+      .map(e => {
+        return {
+          ...e,
+          fieldName: props[fields].find(f => f.id === e[field])?.name ?? '',
+        };
+      })
+      //if 2 transactions share a date and fieldname then sort by UUID
+      .sort((a, b) => a.id.localeCompare(b.id))
+      //secondary: date
+      .sort((a, b) => a.date.localeCompare(b.date))
+      //primary field
+      .sort((a, b) => a.fieldName.localeCompare(b.fieldName));
+
+    return ascDesc === 'desc' ? sortDesc : sortDesc.reverse();
+  }
+
+  function sortCat() {
+    let sortDesc = transactions
+      .map(e => {
+        return {
+          ...e,
+          fieldName: category.find(f => f.id === e[field])?.name ?? '',
+        };
+      })
+      //if 2 transactions share a date and fieldname then sort by UUID
+      .sort((a, b) => a.id.localeCompare(b.id))
+      //secondary: date
+      .sort((a, b) => a.date.localeCompare(b.date))
+      //primary field
+      .sort((a, b) => a.fieldName.localeCompare(b.fieldName)).sort((a, b) => a.id - b.id);
+
+    return ascDesc === 'desc' ? sortDesc : sortDesc.reverse();
```

Comment:
Only problem is that they don't all default to sorting the same way. Default sort for Date is most recent at the top and default for any words is "A" at the top. I've changed the functionality a bit so this may be outdated.